source/hic_hal/stm32/stm32h743ii/extended_features/source/adapter_adc.c

```diff
@@ -0,0 +1,101 @@
+#include "adapter_adc.h"
+#include "stm32h743xx.h"
+#include "stm32h7xx_hal.h"
+
+static __IO uint16_t s_flex_hotplug_adc_value;
+static ADC_HandleTypeDef s_adc_handle;
+static ADC_ChannelConfTypeDef s_chan_conf;
+
+static void error_handler(void)
+{
+    while (1)
+    {
+    }
```

Comment:
since your next task is looked at asserts and crash info, I think asserting on errors might be more informative.

Reply:
I would add a TODO comment for now. I will add crash info after I finish the logging.